vite.config.ts

```diff
@@ -1,108 +1,142 @@
-import { fileURLToPath, URL } from 'node:url';
-import { defineConfig, loadEnv, UserConfig } from 'vite';
-import laravel from 'laravel-vite-plugin';
-import vue from '@vitejs/plugin-vue';
-import i18n from 'laravel-vue-i18n/vite';
+import { fileURLToPath, URL } from "node:url";
+import { defineConfig, loadEnv, PluginOption, UserConfig } from "vite";
+import laravel from "laravel-vite-plugin";
+import vue from "@vitejs/plugin-vue";
+import i18n from "laravel-vue-i18n/vite";
 
 const laravelPlugin = laravel({
-    input: [
-      'resources/sass/app.scss',
-      'resources/js/app.ts',
-    ],
-    refresh: true,
-  })
+	input: ["resources/sass/app.scss", "resources/js/app.ts"],
+	refresh: true,
+});
 
-const baseConfig =   {
-  base: './',
-  plugins: [
-    vue(
-    {
-      template: {
-        transformAssetUrls: {
-          base: null,
-          includeAbsolute: false,
-        },
-      },
-    }),
-    i18n(),
-  ],
-  server: {
-    watch: {
-      ignored: [
-        "**/.*/**",
-        "**/app/**",
-        "**/database/**",
-        "**/node_modules/**",
-        "**/public/**",
-        "**/storage/**",
-        "**/tests/**",
-        "**/vendor/**",
-        "**/presets/**",
-      ],
-    },
-  },
-  resolve: {
-    alias: {
-      // @ts-ignore-next-line
-      '@': fileURLToPath(new URL('./resources/js/', import.meta.url)),
-      vue: 'vue/dist/vue.esm-bundler.js',
-    },
-  },
-  build: {
-    rollupOptions: {
-        output:{
-            manualChunks(id) {
-                if (id.includes('node_modules')) {
-                    return id.toString().split('node_modules/')[1].split('/')[0].toString();
-                }
-            }
-        }
-    }
+const localDevelopMiddleware: PluginOption = {
+  name: "develop-rewrite-middleware",
+  configureServer(server) {
+    const viteIndexPath = "/vite/index.html";
+
+    server.middlewares.use((req, res, next) => {
+      if (!req.url) {
+        next();
+        return;
+      }
+
+      const requestUrl : string = req.url;
+      const doNotTouch = ["/@", "/api", "/resources", "/node_modules"];
+      if (doNotTouch.some((page) => requestUrl.startsWith(page))) {
+        next();
+        return;
+      }
+
+      console.log(req.url);
```

Comment:
Do we want the debug code?

Reply:
better safe on that one, it will only be visible if using local dev without the composer part.